docs/2018-10-24-Configuration-Splitting-Simplification.md

```diff
@@ -0,0 +1,88 @@
+Configuration Splitting Simplification
+======================================
+
+Background
+----------
+
+When doing one-path reachability proofs and applying an axiom,
+`∀ Z . α(Z) → ● β(Z)`, where `●` is the strong-next symbol (aka 'next'),
+to the following reachability goal
+```
+∀ X . φ(X) → ◆ ∃ Y . ψ(X, Y)
+```
+the result will look something like
+```
+￼(∀ X . Φ'(X) → ◆ ∃ Y . ψ(X, Y))
+￼∧
+￼(∀ X . Φ(X) ∧ (¬ ∃ Z. α(Z)) → ● ◆ ∃ Y . ψ(X, Y))
+```
+where `Φ` is either `φ` or `φ` and-ed with something else.
+See `one-path-reachability-proofs.md` for details.
+
+Problem
+-------
+
+We want to rewrite `φ(X) ∧ (¬ ∃ Z. α(Z))` part of the pattern above
+to something more manageable, preferably something that does not use `not`
+and `exists`, except in cases where it can be handled by an SMT solver.
+
+We will assume that
+1. `φ(X) = t(X) ∧ p(X)` where `p(X)` is a predicate, and `t(X)` is a
+   function-like term. It may or may not unify with `α(Z)`.
+1. `α(Z) = s(Z) ∧ q(X)` where `s(Z)` is a functional term, composed out of
+    constructor-like symbols and variables and `q(x)` is a predicate
+    (so `α(Z)` is function-like).
+
+The proposed solution should not return a broken result when these assumptions
+do not hold, but it is not forced to fully simplify the given term.
+
+Solution
+--------
+
+```
+φ(X) ∧ (¬ ∃ Z. α(Z)) = t(X) ∧ p(X) ∧ (¬ ∃ Z. α(Z))
+```
+We will examine only the `t(X) ∧ (¬ ∃ Z. α(Z))` term next.
+```
+t(X) ∧ (¬ ∃ Z. α(Z))
+    = t(X) ∧ ⌈t(X) ∧ (¬ ∃ Z. α(Z))⌉
```

Comment:
a little explanation of the steps would be nice. This first line is justified because `t(X)` is (or should be) assumed functional above, right?